draft-add-requirements.md

```diff
@@ -142,18 +271,97 @@ Narrative required.
+
+- Prevent devices along the network path to the content-provider's resolver from observing client DNS messages
+- Verify that answers come from the entity that manages the domains being resolved
+- Reduce the number of entities able to monitor the specific names accessed by a client to only the client and the content provider, assuming that the content provider would already see the names upon a secure connection later being made based on the DNS answers (e.g., in the TLS SNI extension)
 
 | Requirement | Description |
 | R12.1 | Example requirement |
 
+# Privacy and security requirements {#priv-sec}
+
+Encrypted DNS improves the privacy and security of DNS queries and answers in the presence of malicious
+attackers. Such attackers are assumed to interfere with or otherwise impede DNS traffic and corresponding
+discovery mechanisms. They may be on-path or off-path between the client and entities with which the client
+communicates {{?RFC3552}}. These attackers can inject, tamper, or otherwise interfere with traffic as needed.
+Given these capabilities, an attacker may have a variety of goals, including, though not limited to:
+
+- Monitor and profile clients by observing unencrypted DNS traffic
+
+- Modify unencrypted DNS traffic to filter or augment the user experience
+
+- Block encrypted DNS
+
+Clients cannot assume that their network does not have such an attacker unless given some means of authenticating or otherwise
+trusting the communication with their DNS resolver.
+
+Given this type of attacker, resolver discovery mechanisms must be designed carefully to not worsen a client's security or
+privacy posture. In particular, attackers must not be able to:
+
+- Redirect secure DNS traffic to themselves when they would not otherwise handle DNS traffic.
+
+- Override or interfere with the resolver preferences of a user or administrator.
+
+- Cause clients to use a discovered resolver which has no authenticated delegation from a client-known entity.
+
+- Influence automatic discovery mechanisms such that a client uses one or more resolvers that are not
+otherwise involved with providing service to the client, such as: a network provider, a VPN server, a
+content provider being accessed, or a server that the client has manually configured.
+
+Beyond these requirements, standards describing resolver discovery mechanisms must not place any requirements
+on clients to select particular resolvers over others.
+
+## On opportunistic encryption
+
+Opportunistic encrypted DNS, when the client cannot authenticate the entity that provides encrypted DNS, does
+not meet the requirements laid out here for resolver discovery. While opportunistic encryption can provide some
+benefits, specifically in reducing the ability for other entities to observe traffic, it is not a viable solution
+against an on-path attacker.
+
+Performing opportunistic encrypted DNS does not require specific discovery mechanisms. Section 4.1 of {{?RFC7858}}
```

Comment:
Opportunistic encryption is specific to DoT and DoH does not discuss Opportunistic encryption. This is a question to the WG whether all DNS encryption protocols should support opportunistic encryption profile ?

Reply:
I think this PR is the wrong place to ask a question to the WG. If you asked the WG I suspect the answer would be that it's not worth developing opportunistic for DoH due to the lack of benefit and the fact that if you want to do this you can just use DoT like Android does.

Reply:
Agreed but the text should say opportunistic encryption is specific to DoT and DoH does not support opportunistic encryption. The current text in this paragraph is assuming DNS traffic is encrypted, it is not correct. Opportunistic encryption allows the client to fallback to Do53. You may want to look into https://tools.ietf.org/html/rfc8310 

<snip from our spec>
  In the best-case scenario for the
   Opportunistic Privacy profile (an authenticated and encrypted
   connection), it is equivalent to the Strict Privacy profile.  In the
   worst-case scenario, it is equivalent to cleartext.  Clients using
   the Opportunistic Privacy profile SHOULD try for the best case but
   MAY fall back to the intermediate case and, eventually, the worst-
   case scenario, in order to obtain a response.
</snip>

Reply:
I think we need to leave this to be a generic statement about opportunistic encryption overall.

Reply:
My comment is about the the following line in the draft:

"While opportunistic encryption can provide some benefits, specifically in reducing the ability for other entities to observe traffic, it is not a viable solution against an on-path attacker."

The above line is not correct, opportunistic allows the client to fallback to clear text DNS (you may want to look into https://tools.ietf.org/html/rfc8310 for more details)..

Reply:
The fact that it fails open to unencrypted clear text is precisely one of the reasons it is not a viable solution against an on-path attacker.